virttest/utils_libvirt/libvirt_vmxml.py

```diff
@@ -65,3 +74,55 @@ def remove_vm_devices_by_type(vm, device_type):
 
     if vm_was_running:
         vm.start()
+
+
+def create_vm_device_by_type(dev_type, dev_dict):
+    """
+    Create device by device type
+
+    :param dev_type: device type
+    :param dev_dict: dict for device
+    :return: device object
+    """
+    dev_obj = None
+    if dev_type == 'disk':
+        dev_obj = disk.Disk()
+    elif dev_type == 'controller':
+        dev_obj = controller.Controller()
+    elif dev_type == 'interface':
+        dev_obj = interface.Interface()
+    elif dev_type == 'channel':
+        dev_obj = channel.Channel()
+    elif dev_type == 'vsock':
+        dev_obj = vsock.Vsock()
+    elif dev_type == 'watchdog':
+        dev_obj = watchdog.Watchdog()
+    elif dev_type == 'rng':
+        dev_obj = rng.Rng()
+    elif dev_type == 'hostdev':
+        dev_obj = hostdev.Hostdev()
+
+    dev_obj.setup_attrs(**dev_dict)
+
+    return dev_obj
+
+
+def get_or_create_vm_device_if_absent(vmxml, dev_type, dev_dict, index=0):
```

Comment:
After further thinking, this function's purpose should be updating a vm device using an unified way instead of creating multiple updating functions for different devices. If the device exists, let's update it with given dev_dict, otherwise, let's create it with given dev_dict. 
So shall we rename it as 'modify_vm_device'? @chunfuwen @Yingshun @chloerh

Reply:
I agree with you. It's really weird returning the existing device without any updates or comparisons in this function. Then, I think it'll be much more clear if the device will be updated with the given dev_dict.